datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -107,9 +127,102 @@ func (f *ChainDataFetcher) Stop() error {
 	logger.Info("wait for all goroutines to be terminated...")
 	f.wg.Wait()
 	logger.Info("terminated all goroutines for chaindatafetcher")
+	f.started = false
+	return nil
+}
+
+func (f *ChainDataFetcher) start() error {
+	f.mu.Lock()
+	defer f.mu.Unlock()
+	if f.started {
+		return errors.New("the chaindata fetcher is already started")
+	}
+
+	f.chainSub = f.blockchain.SubscribeChainEvent(f.chainCh)
+	// TODO-ChainDataFetcher add logic to request from checkpoint to the first posted event after start.
+
+	f.started = true
 	return nil
 }
 
+func (f *ChainDataFetcher) stop() error {
+	f.mu.Lock()
+	defer f.mu.Unlock()
+	if !f.started {
+		return errors.New("the chaindata fetcher is already stopped")
```

Comment:
Below would be more appropriate.
```suggestion
		return errors.New("the chaindata fetcher is not running")
```

Reply:
Modified on 225a181fc813970d4525c5ced9664518b7889f07.